http/beaconstate.go

```diff
@@ -70,31 +71,66 @@ func (s *Service) beaconStateFromSSZ(res *httpResponse) (*api.Response[*spec.Ver
 		Metadata: metadataFromHeaders(res.headers),
 	}
 
+	var dynSSZ *dynssz.DynSsz
+	if s.useDynamicSSZ {
+		dynSSZ = dynssz.NewDynSsz(s.spec)
+	}
+
 	switch res.consensusVersion {
 	case spec.DataVersionPhase0:
 		response.Data.Phase0 = &phase0.BeaconState{}
-		if err := response.Data.Phase0.UnmarshalSSZ(res.body); err != nil {
-			return nil, errors.Join(errors.New("failed to decode phase0 beacon state"), err)
+		if s.useDynamicSSZ {
+			if err := dynSSZ.UnmarshalSSZ(response.Data.Phase0, res.body); err != nil {
+				return nil, errors.Join(errors.New("failed to dynamic decode phase0 beacon state"), err)
+			}
+		} else {
+			if err := response.Data.Phase0.UnmarshalSSZ(res.body); err != nil {
+				return nil, errors.Join(errors.New("failed to decode phase0 beacon state"), err)
+			}
 		}
 	case spec.DataVersionAltair:
 		response.Data.Altair = &altair.BeaconState{}
-		if err := response.Data.Altair.UnmarshalSSZ(res.body); err != nil {
-			return nil, errors.Join(errors.New("failed to decode altair beacon state"), err)
+		if s.useDynamicSSZ {
+			if err := dynSSZ.UnmarshalSSZ(response.Data.Altair, res.body); err != nil {
+				return nil, errors.Join(errors.New("failed to dynamic decode altair beacon state"), err)
+			}
+		} else {
+			if err := response.Data.Altair.UnmarshalSSZ(res.body); err != nil {
+				return nil, errors.Join(errors.New("failed to decode altair beacon state"), err)
+			}
```

Comment:
Not a massive fan of further nesting control structures here.  Perhaps something like:

```
var err error
if s.useDynamicSSZ {
    err = dynSSZ.UnmarshalSSZ(response.Data.Altair, res.body)
} else {
    err = response.Data.Altair.UnmarshalSSZ(res.body)
}
if err != nil {
    return nil, errors.Join(errors.New("failed to decode altair beacon state"), err)
}
```

Reply:
looks cleaner, yea :)
changed in signedbeaconblock.go as well